xdsl/printer.py

```diff
@@ -349,7 +348,27 @@ def print_paramattr_parameters(
         self.print(">")
 
     def print_string_literal(self, string: str):
-        self.print(json.dumps(string))
+        # Common case, this makes a huge difference because we print one time instead
+        # of once per character.
+        # This makes a huge difference on programs with a lot of generic syntax.
+        if all(
+            (0x20 <= ord(char) <= 0x7E) and char not in ('"', "\\") for char in string
+        ):
+            self.print(f'"{string}"')
+            return
+        hexdigits = "0123456789ABCDEF"
+        self.print('"')
+        for char in string:
+            o = ord(char)
+            if char == "\\":
+                self.print("\\\\")
```

Comment:
I think it makes it easier to look at non-escaped strings to use `r"\\"` here (the `r` removes all escape sequences, it's usually used for writing regexes).

But opinions may vary here, and pyright, ruff, or whatever else we may use might disagree.

Reply:
Definitely agree, whatever removes some escape hell layers, thanks for the idea!